src/reporting/clients/ODataClient.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { AccessToken } from "@itwin/core-bentley";
+import { ODataResponse, ODataItem, ODataEntityResponse, ODataEntityValue } from "../interfaces/OData";
+import { OperationsBase } from "../OperationsBase";
+
+export interface ODataClientInterface{
+  getODataReport(
+    accessToken: AccessToken,
+    reportId: string
+  ): Promise<ODataResponse>,
+  getODataReportEntity(
+    accessToken: AccessToken,
+    reportId: string,
+    odataItem: ODataItem
+  ): Promise<ODataEntityValue | undefined>,
+  getODataReportMetadata(
+    accessToken: AccessToken,
+    reportId: string
+  ): Promise<Response>,
+}
+
+export class ODataClient extends OperationsBase implements ODataClientInterface{
+  /**
+   * Lists all OData Entities for a Report.
+   * @param {string} reportId The Report Id.
+   * @param {string} accessToken OAuth access token with scope `insights:read`
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/odata/
+   */
+  public async getODataReport(accessToken: AccessToken, reportId: string): Promise<ODataResponse> {
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return this.fetchData<ODataResponse>(url, requestOptions);
+  }
+
+  /**
+   * Lists the raw table data for a Report Entity.
+   * @param {string} reportId The Report Id.
+   * @param {ODataItem} odataItem Reference to a table exported to your Report. Use {@link getODataReport()} to fetch a list of ODataItems in the report.
+   * @param {string} accessToken OAuth access token with scope `insights:read`
+   * @memberof ReportingClient
+   * @link https://developer.bentley.com/apis/insights/operations/odata-entity/
+   */
+  public async getODataReportEntity(accessToken: AccessToken, reportId: string, odataItem: ODataItem) {
+    const segments = odataItem.url.split('/');  // region, manifestId, entityType
+    if (segments.length !== 3) {
+      return undefined;
+    }
+    let sequence = 0;
+    const reportData: ODataEntityValue = [];
+    let response: ODataEntityResponse;
+    let url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/${encodeURIComponent(segments[0])}/${encodeURIComponent(segments[1])}/${encodeURIComponent(segments[2])}?sequence=${encodeURIComponent(sequence)}`;
```

Comment:
This can be simplified to `${this.basePath}/odata/${encodeURIComponent(reportId)}/${odataItem.url}`.